src/librepgp/validate.c

```diff
@@ -969,5 +848,85 @@ pgp_validate_mem(pgp_io_t *             io,
         pgp_memory_free(validation.mem);
     }
 
-    return validate_result_status(io->errs, NULL, result);
+    return validate_result_status(NULL, result);
+}
+
+bool
+validate_result_status(const char *f, pgp_validation_t *val)
+{
+    time_t now;
+    time_t t;
+    char   buf[128];
+
+    now = time(NULL);
+    if (now < val->birthtime) {
+        /* signature is not valid yet! */
+        if (f) {
+            (void) fprintf(stderr, "\"%s\": ", f);
+        } else {
+            (void) fprintf(stderr, "memory ");
+        }
+        (void) fprintf(stderr,
+                       "signature not valid until %.24s (%s)\n",
+                       ctime(&val->birthtime),
+                       fmtsecs((int64_t)(val->birthtime - now), buf, sizeof(buf)));
+        return false;
+    }
+    if (val->duration != 0 && now > val->birthtime + val->duration) {
+        /* signature has expired */
+        t = val->duration + val->birthtime;
+        if (f) {
+            (void) fprintf(stderr, "\"%s\": ", f);
+        } else {
+            (void) fprintf(stderr, "memory ");
+        }
+        (void) fprintf(stderr,
+                       "signature not valid after %.24s (%s ago)\n",
+                       ctime(&t),
+                       fmtsecs((int64_t)(now - t), buf, sizeof(buf)));
+        return false;
+    }
+    return val->validc && !val->invalidc && !val->unknownc;
+}
+
+bool
+pgp_validate_key_sigs(pgp_validation_t *     result,
+                      const pgp_key_t *      key,
+                      const rnp_key_store_t *keyring,
+                      pgp_cb_ret_t cb_get_passphrase(const pgp_packet_t *, pgp_cbdata_t *))
+{
+    pgp_stream_t *    stream;
+    validate_key_cb_t keysigs;
+
+    (void) memset(&keysigs, 0x0, sizeof(keysigs));
+    keysigs.result = result;
+    keysigs.getpassphrase = cb_get_passphrase;
+
+    stream = pgp_new(sizeof(*stream));
```

Comment:
may `pgp_new` return `NULL`?

Reply:
This code was actually moved from some other place. But keep going, it's good opportunity to fix old code as well.

Reply:
@catap OK with the fix?

Reply:
@ronaldtse yes